packages/garbo/src/tasks/barfTurn.ts

```diff
@@ -340,6 +342,30 @@ const NonBarfTurnTasks: AlternateTask[] = [
     spendsTurn: true,
     choices: { 1091: 7 },
   },
+  {
+    name: "Saxophone Lucky",
+    completed: () => !AprilingBandHelmet.canPlay("Apriling band saxophone"),
+    ready: () =>
+      have($item`Apriling band saxophone`) &&
+      getBestLuckyAdventure().location ===
+        $location`The Castle in the Clouds in the Sky (Top Floor)` &&
```

Comment:
I don't see a reason to check .location against the top floor? this won't work as soon as another location is added

Reply:
why would we bother having separate tasks when here we only would care about the single most profitable adventure that isn't embezzlers? can't this just be generic to be able to handle whichever location is returned

Reply:
What I'd do is add a parameter to `LuckyAdventure` that describes what phase of garbo we use it in--`yachtzee` for the eventual fishy, `embezzler` for embezzlers, and `barf` for this, and do `getBestLuckyAdventure()?.phase === "barf"` or whatever. I'd also make it explicitly return null when the list of viable options is empty, rather than not filtering the list and always using the maxBy. But I guess that can be problematic, and it gets annoying to tell grimoire to adventure in a place that might not exist.

Reply:
Definitely do the first part tho